ruby_sinatra/server.rb

```diff
@@ -1,8 +1,38 @@
 require 'bundler/setup'
 require 'sinatra'
+require 'mysql2'
+require 'json'
 
 set :port, 6969
 
+def db
+  @db ||= Mysql2::Client.new({
+    host: "localhost",
+    username: ENV["EBOSHI_API_SHOOTOUT_MYSQL_USERNAME"] || "root",
+    password: ENV["EBOSHI_API_SHOOTOUT_MYSQL_PASSWORD"] || "",
+    database: ENV["EBOSHI_API_SHOOTOUT_MYSQL_DATABASE"] || "eboshi_test",
+    database_timezone: :utc,
+  })
+end
+
 get '/api/test' do
   'Hello world'
 end
+
+get '/api/clients' do
+  @clients = db.query("SELECT * FROM clients")
+
+  response = { data: [] }
+  @clients.each do |client|
```

Comment:
get real familiar with http://ruby-doc.org/core-2.2.3/Enumerable.html . these ideas are in every language, including javascript, so this will help you no matter what you end up programming in.